apps/condo/domains/contact/components/ContactsEditor.tsx

```diff
@@ -0,0 +1,556 @@
+    // Refs are used to keep it
+
+    const contactFieldsRef = useRef(contactFields)
+    useEffect(() => {
+        contactFieldsRef.current = contactFields
+    }, [contactFields])
+
+    const shouldCreateContactRef = useRef(shouldCreateContact)
+    useEffect(() => {
+        shouldCreateContactRef.current = shouldCreateContact
+    }, [shouldCreateContact])
+
+    const createContactAction = Contact.useCreate({ }, () => Promise.resolve())
+
+    const handleChangeContact = (values, isNew) => {
+        setContactFields(values)
+        setShouldCreateContact(isNew)
+    }
+
+    const createContact = async (organization, property, unitName) => {
+        if (shouldCreateContactRef.current) {
+            try {
+                return await createContactAction({
+                    ...contactFieldsRef.current,
+                    organization,
+                    property,
+                    unitName,
+                })
+            } catch (e) {
+                // Duplicated contacts should be figured out on the client,
+                // and "create" action should not be performed.
+                // In case of violation of unique constraint on `Contact` table,
+                // be silent for a user, but make a record in log.
+                if (e.message.match('Contact_uniq')) {
+                    console.error(e)
+                } else {
+                    throw (e)
+                }
+            }
+        }
+    }
+
+    const ContactsEditorComponent: React.FC<IContactEditorProps> = useMemo(() => {
+        const ContactsEditorWrapper = (props) => (
+            <ContactsEditor
+                {...props}
+                organization={organization}
+                onChange={handleChangeContact}
+            />
+        )
+        return ContactsEditorWrapper
+    }, [])
+
+    return {
+        createContact,
+        ContactsEditorComponent,
+    }
+}
+
+export const ContactsEditor: React.FC<IContactEditorProps> = (props) => {
+    const intl = useIntl()
+    const FullNameLabel = intl.formatMessage({ id: 'contact.Contact.ContactsEditor.Name' })
+    const PhoneLabel = intl.formatMessage({ id: 'contact.Contact.ContactsEditor.Phone' })
+    const AddNewContactLabel = intl.formatMessage({ id: 'contact.Contact.ContactsEditor.AddNewContact' })
+    const AnotherContactLabel = intl.formatMessage({ id: 'contact.Contact.ContactsEditor.AnotherContact' })
+    const CannotCreateContactMessage = intl.formatMessage({ id: 'contact.Contact.ContactsEditor.CannotCreateContact' })
+
+    const { form, fields, value: initialValue, onChange, organization, property, unitName } = props
+
+    const [contacts, setContacts] = useState([])
+    const [loading, setLoading] = useState(true)
+    const [error, setError] = useState()
+    const { link: { role } } = useOrganization()
+    const client = useApolloClient()
+
+    searchContacts(client, {
```

Comment:
call this onMount

Reply:
I'm sorry, I don't understand `onMount` of what should this function be called and why?